styx-api-service/src/main/java/com/spotify/styx/api/SchedulerProxyResource.java

```diff
@@ -48,12 +51,22 @@ public class SchedulerProxyResource {
 
   private final String schedulerServiceBaseUrl;
   private final Client client;
+  private final String schedulerHost;
 
   public SchedulerProxyResource(String schedulerServiceBaseUrl, Client client) {
     this.schedulerServiceBaseUrl = Objects.requireNonNull(schedulerServiceBaseUrl);
+    this.schedulerHost = getHost(schedulerServiceBaseUrl);
     this.client = Objects.requireNonNull(client, "client");
   }
 
+  private String getHost(String schedulerServiceBaseUrl) {
+    try {
+      return new URL(schedulerServiceBaseUrl).getHost();
```

Comment:
Let's use `URI.create(schedulerServiceBaseUrl)` to save this exception wrapping.